pyclaragenomics/setup_pyclaragenomics.py

```diff
@@ -0,0 +1,105 @@
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+import argparse
+import os.path
+import os
+import subprocess
+
+
+def get_relative_path(sub_folder_name):
+    return os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        sub_folder_name
+    )
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='build & install Clara Genomics Analysis SDK.')
+    parser.add_argument('--build_output_folder',
+                        required=False,
+                        default=get_relative_path("cga_build"),
+                        help="Choose an output folder for building")
+    parser.add_argument('--develop',
+                        required=False,
+                        action='store_true',
+                        help="CInstall using pip editble mode")
+    return parser.parse_args()
+
+
+class CMakeWrapper:
+    """Class to encapsulate building a CMake project."""
+
+    def __init__(self, cmake_root_dir, pycga_dir, cmake_build_path="cmake_build",
+                 cmake_extra_args="", develop_mode=False):
+        """
+        Class constructor.
+
+        Args:
+            cmake_root_dir : Root directory of CMake project
+            pycga_dir : pyclaragenomics folder location
+            cmake_build_path : cmake build output folder
+            cmake_extra_args : Extra string arguments to be passed to CMake during setup
+            develop_mode: to setup pyclaragenomics in development mode
+        """
+        self.cmake_root_dir = os.path.abspath(cmake_root_dir)
+        self.pycga_dir = os.path.abspath(pycga_dir)
+        self.build_path = os.path.abspath(cmake_build_path)
+        self.cga_install_dir = os.path.join(self.build_path, "install")
+        self.cmake_extra_args = cmake_extra_args
+        self.develop_mode = develop_mode
+        self.cuda_toolkit_root_dir = os.environ.get("CUDA_TOOLKIT_ROOT_DIR")
+
+    def run_cmake_cmd(self):
+        cmake_args = ['-DCMAKE_INSTALL_PREFIX=' + self.cga_install_dir,
+                      '-DCMAKE_BUILD_TYPE=' + 'Release',
+                      '-DCMAKE_INSTALL_RPATH=' + os.path.join(self.cga_install_dir, "lib")]
+        cmake_args += [self.cmake_extra_args]
+
+        if self.cuda_toolkit_root_dir:
+            cmake_args += ["-DCUDA_TOOLKIT_ROOT_DIR=%s" % self.cuda_toolkit_root_dir]
+
+        if not os.path.exists(self.build_path):
+            os.makedirs(self.build_path)
+
+        subprocess.check_call(['cmake', self.cmake_root_dir] + cmake_args, cwd=self.build_path)
+
+    def run_build_cmd(self):
+        build_args = ['--', '-j16', 'install']
+        subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_path)
+
+    def setup_python_binding(self):
```

Comment:
Let's move this into a separate function outside this class since this function doesn't have anything to do with make wrapping. And this function can then take in arguments we get from the CMakeWrapper class